src/test/java/edu/kis/powp/jobs2d/events/MacroStopListener.java

```diff
@@ -1,13 +1,25 @@
 package edu.kis.powp.jobs2d.events;
 
+import edu.kis.powp.jobs2d.drivers.DriverManager;
+import edu.kis.powp.jobs2d.drivers.MacroDecoratorDriver;
 import edu.kis.powp.jobs2d.drivers.MacroRecorder;
 
 import java.awt.event.ActionEvent;
 import java.awt.event.ActionListener;
 
 public class MacroStopListener implements ActionListener {
+    private DriverManager driverManager;
+    public MacroStopListener(DriverManager driverManager) {
+        this.driverManager = driverManager;
+    }
     @Override
     public void actionPerformed(ActionEvent e) {
         MacroRecorder.stopRecording();
+        try {
+            MacroDecoratorDriver mDriver = (MacroDecoratorDriver) driverManager.getCurrentDriver();
+            driverManager.setCurrentDriver(mDriver.getDriver());
+        } catch (ClassCastException exp) {
+            //"Macro driver changed"
```

Comment:
It would be best to provide users with info that something went wrong (e.g. logger error info)